org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/AnalyzeProjectJob.java

```diff
@@ -185,39 +185,24 @@ private void runAnalysisAndUpdateMarkers(final IProgressMonitor monitor, IProjec
     }
 
     Map<IResource, List<Issue>> issuesPerResource = new LinkedHashMap<>();
-    request.getFiles().forEach(file -> issuesPerResource.put(file, new ArrayList<>()));
+    request.getFiles().forEach(fileWithDoc -> issuesPerResource.put(fileWithDoc.getFile(), new ArrayList<>()));
 
     AnalysisResults result = runAndCheckCancellation(config, sonarProject, issuesPerResource, monitor);
     if (!monitor.isCanceled() && result != null) {
-      updateMarkers(issuesPerResource, result, request.getTriggerType());
+      updateMarkers(docPerFiles, issuesPerResource, result, request.getTriggerType());
     }
   }
 
-  private static List<ClientInputFile> buildInputFiles(final List<IFile> filesToAnalyze, Map<IFile, String> fileLanguages, IProgressMonitor monitor) {
+  private static List<ClientInputFile> buildInputFiles(final Map<IFile, IDocument> filesToAnalyze, Map<IFile, String> fileLanguages, IProgressMonitor monitor) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "monitor". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1172)